src/MoBi.Presentation/Presenter/SelectNeighborPathPresenter.cs

```diff
@@ -50,19 +55,33 @@ private void onSelectedContainerPathChanged(IEntity entity, ObjectPath container
          ViewChanged();
       }
 
-      public void Init(SpatialStructure spatialStructure, string label, string defaultSelection = null)
+      public void Init(string label, string defaultSelection = null)
       {
          _view.Label = label;
-         var organism = spatialStructure.TopContainers.Find(x => x.ContainerType == ContainerType.Organism) ?? spatialStructure.TopContainers.FirstOrDefault();
-
+         
+         var topContainers = _buildingBlockRepository.SpatialStructureCollection.SelectMany(mapTopContainerDTOs).ToList();
+         
          //no organism found, nothing to do?
-         if (organism == null)
+         if (!topContainers.Any())
             return;
 
-         _selectContainerInTreePresenter.InitTreeStructure(new[] {_containerDTOMapper.MapFrom(organism)});
+         _selectContainerInTreePresenter.InitTreeStructure(topContainers);
          _selectedPathDTO.Path = defaultSelection ?? string.Empty;
       }
 
+      private IReadOnlyList<ObjectBaseDTO> mapTopContainerDTOs(SpatialStructure spatialStructure)
+      {
+         var containers = spatialStructure.TopContainers.Where(x => x.ContainerType == ContainerType.Organism).ToList();
+         if (!containers.Any())
```

Comment:
I'm trying to reproduce the previous logic, which was
1. try to find the first 'Organism' in top containers
2. if none, just use the first top container

I changed it a little because I don't know why it's limited to only a single top container. The building block could have more than one and we are changing this display to show top containers from multiple modules anyway.